unikernel.ml

```diff
@@ -779,6 +779,78 @@ struct
                 ~icon:"/images/robur.png" ())
              `Not_found)
 
+  let edit_policy albatross store uuid reqd (user : User_model.user) =
+    let users = User_model.create_user_uuid_map (snd store).Storage.users in
+    match User_model.find_user_by_key uuid users with
+    | Some u ->
+        let user_policy =
+          match Albatross.policy albatross ~domain:u.name with
+          | Ok p -> (
+              match p with Some p -> p | None -> Albatross.empty_policy)
+          | Error _ -> Albatross.empty_policy
+        in
+        let policy_avalaible = Albatross.policy_resource_avalaible albatross in
+        Lwt.return
+          (reply reqd ~content_type:"text/html"
+             (Dashboard.dashboard_layout user
+                ~page_title:(String.capitalize_ascii u.name ^ " | Mollymawk")
+                ~content:
+                  (Update_policy.update_policy_layout u ~user_policy
+                     ~root_policy:policy_avalaible)
+                ~icon:"/images/robur.png" ())
+             `OK)
+    | None ->
+        let status =
+          {
+            Utils.Status.code = 404;
+            title = "Error";
+            data = "Couldn't find account with uuid: " ^ uuid;
+            success = false;
+          }
+        in
+        Lwt.return
+          (reply reqd ~content_type:"text/html"
+             (Guest_layout.guest_layout ~page_title:"404 | Mollymawk"
+                ~content:(Error_page.error_layout status)
+                ~icon:"/images/robur.png" ())
+             `Not_found)
+
+  let update_policy store albatross reqd _user =
+    decode_request_body reqd >>= fun data ->
+    let json =
+      try Ok (Yojson.Basic.from_string data)
+      with Yojson.Json_error s -> Error (`Msg s)
+    in
+    match json with
+    | Error (`Msg err) ->
+        Logs.warn (fun m -> m "Failed to parse JSON: %s" err);
+        http_response reqd ~title:"Error" ~data:(String.escaped err)
+          `Bad_request
+    | Ok json -> (
+        let user_uuid =
+          Yojson.Basic.(to_string (json |> Util.member "user_uuid"))
+        in
+        let users = User_model.create_user_uuid_map (snd store).Storage.users in
+        match User_model.find_user_by_key user_uuid users with
+        | Some u -> (
+            match Albatross_json.policy_of_json json with
+            | Ok policy -> (
```

Comment:
Here I'd expect yet another check:
```
let root_policy = Albatross.policy albatross in
match Vmm_core.Policy.is_smaller ~super:root_policy ~sub:policy with
| Error `Msg msg -> log an error and return the error message
| Ok () -> .. as here
```

This way we ensure that the policy is usable and doesn't contradict the root policy (e.g. only cpuids that are available are assigned, ...).